MapboxSearch/offline/src/main/java/com/mapbox/search/offline/OfflineSearchEngine.kt

```diff
@@ -205,6 +205,46 @@ public interface OfflineSearchEngine {
         callback = callback,
     )
 
+    /**
+     * Performs a search along the supplied [route].
+     *
+     * @param query the search query
+     * @param proximity coordinate along the route
+     * @param route list of points that make up route line
+     * @param executor executor for dispatching event, by default events are dispatched to the main thread
+     * @param callback search result callback, delivers results on the main thread
+     * @return [AsyncOperationTask] object which allows to cancel the request.
+     */
+    public fun searchAlongRoute(
```

Comment:
I believe such functionality would be well-suited for a separate module, something like `search-sdk-extensions`. It's crucial to keep the core functionality as lightweight as possible. Adding a new dependency like Mapbox Turf  SDK could slightly increase the overall Offline Search SDK binary size for all the customers, even those who doesn't need `searchAlongRoute` functionality. This might be a concern for some customers.

Reply:
Thank you @DzmitryFomchyn. I'll evaluate this idea.

Reply:
Speaking with @mapbox-danny, we're okay to continue with the Mapbox Turf as the size is relatively small.